src/pkg/services/m365/api/users.go

```diff
@@ -229,42 +227,33 @@ func (c Users) GetIDAndName(ctx context.Context, userID string) (string, string,
 func (c Users) GetInfo(ctx context.Context, userID string) (*UserInfo, error) {
 	// Assume all services are enabled
 	// then filter down to only services the user has enabled
-	var (
-		err      error
-		userInfo = newUserInfo()
+	userInfo := newUserInfo()
 
-		requestParameters = &users.ItemMailFoldersRequestBuilderGetQueryParameters{
-			Select: []string{"id"},
-			Top:    ptr.To[int32](1), // if we get any folders, then we have access.
-		}
+	requestParameters := users.ItemMailFoldersRequestBuilderGetQueryParameters{
+		Select: []string{"id"},
+		Top:    ptr.To[int32](1), // if we get any folders, then we have access.
+	}
 
-		options = users.ItemMailFoldersRequestBuilderGetRequestConfiguration{
-			QueryParameters: requestParameters,
-		}
-	)
+	options := users.ItemMailFoldersRequestBuilderGetRequestConfiguration{
+		QueryParameters: &requestParameters,
+	}
 
-	userInfo.HasMailBox = true
+	if _, err := c.GetMailFolders(ctx, userID, options); err != nil {
```

Comment:
this is a bit out of the scope of this PR, but will we ever need to check all of the data types for Exchange? I seem to recall some reports of seeing users with no contacts folders but I'm unsure if they had email folders

Reply:
Good question. To the best of my memory, any issue that involved missing "contact folders" was, at base, a failure to get MailFolders.  So I think we're safe?  I believe, since contacts is a well known folder, that microsoft will guarantee its creation at the time of creating mailfolders.